app/models/post.rb

```diff
@@ -0,0 +1,21 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: :author_id
+  has_many :comments
+  has_many :likes
+
+  attribute :title, :text
+  attribute :text, :text
+  attribute :comments_counter, :integer, default: 0
+  attribute :likes_counter, :integer, default: 0
+
+  after_create :increment_author_posts_counter
+
+  def increment_author_posts_counter
+    puts 'Callback invoked!'
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] The `after_save` callback might indeed be a better choice in this scenario, as it's more flexible and versatile compared to `after_create`. Also, marking the method as private would be a good practice to encapsulate the behavior. Here's how you might suggest this:

```ruby
  after_save :increment_author_posts_counter

  private

  def increment_author_posts_counter
    puts 'Callback invoked!'
    author.increment!(:posts_counter)
  end

```

By using `after_save`, you're allowing the same functionality to run after both `create` and `update` actions. This can be useful if you ever need to modify this behavior in the future. Additionally, marking the method as private ensures that it's only accessible from within the class and can't be called externally, providing better encapsulation and maintaining the principle of least privilege.